File: tests/src/org.apache.kafka/kafka-clients/4.2.0/src/test/java/org_apache_kafka/kafka_clients/GeneratedMessageAnonymous4Test.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_kafka.kafka_clients;

import org.apache.kafka.shaded.com.google.protobuf.Descriptors;
import org.apache.kafka.shaded.io.opentelemetry.proto.profiles.v1experimental.Profile;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class GeneratedMessageAnonymous4Test {

    @Test
    void loadsGeneratedDescriptorViaFieldAccessorTable() {
        Descriptors.Descriptor descriptor = Profile.getDescriptor();

        assertThat(descriptor.findFieldByName("time_nanos")).isNotNull();
    }
}

File: tests/src/org.apache.pekko/pekko-protobuf-v3_3/1.0.1/src/test/scala/org_apache_pekko/pekko_protobuf_v3_3/GeneratedMessageAnonymous4Test.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_3

import org.apache.pekko.protobufv3.internal.Descriptors
import org.apache.pekko.protobufv3.internal.GeneratedMessage
import org.apache.pekko.protobufv3.internal.Message
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class GeneratedMessageAnonymous4Test {
  @Test
  def fileScopedExtensionLoadsDescriptorFromGeneratedDescriptorClass(): Unit = {
    val extension: GeneratedMessage.GeneratedExtension[Message, String] =
      GeneratedMessage.newFileScopedGeneratedExtension[Message, String](
        classOf[GeneratedMessageAnonymous4DescriptorProbe],
        null,
        classOf[GeneratedMessageAnonymous4DescriptorProbe].getName,
        "covered_extension")

    val descriptor: Descriptors.FieldDescriptor = extension.getDescriptor

    assertThat(descriptor.getName).isEqualTo("covered_extension")
    assertThat(descriptor.getNumber).isEqualTo(100)
    assertThat(descriptor.getJavaType).isEqualTo(Descriptors.FieldDescriptor.JavaType.STRING)
    assertThat(descriptor.getFile).isSameAs(GeneratedMessageAnonymous4DescriptorProbe.descriptor)
  }
}
